### PR TITLE
Add smart contract link

### PR DESCRIPTION
**Pull request by solovev**

### CODE REVIEW

File: app/earn/[...vault]/header/first-line/vault-title.tsx [hunk 1]
```diff
@@ -1,30 +1,51 @@
...
+}
+
+function HowVaultWorks() {
   return (
-    <div className={styles.container}>
-      {children}
-    </div>
+    <WrapperLink href="https://docs.eonian.finance/basics/how-eonian-works" icon={<IconExternal />} iconAtEnd>
+      How Vault works?
+    </WrapperLink>
```
Comment (LeoVS09) [inline app/earn/[...vault]/header/first-line/vault-title.tsx:18]: It should handle it automatically
```suggestion
    <WrapperLink href="https://docs.eonian.finance/basics/how-eonian-works" withIcon>
      How Vault works?
    </WrapperLink>
```

File: app/earn/[...vault]/header/first-line/vault-title.tsx [hunk 2]
```diff
@@ -1,30 +1,51 @@
...
 
-function HowVaultWorks() {
-  return <ExternalAction href="https://docs.eonian.finance/basics/how-eonian-works">How Vault works?</ExternalAction>
+function ContractAddress() {
+  const { vault, chainId } = useVaultContext()
+  const url = `${getChainExplorer(chainId)}/address/${vault.address}#code`
+  return (
+    <WrapperLink href={url} icon={<IconExternal />} iconAtEnd>
+      Smart Contract
+    </WrapperLink>
```
Comment (LeoVS09) [inline app/earn/[...vault]/header/first-line/vault-title.tsx:28]:
```suggestion
    <WrapperLink href={url} withIcon>
      Smart Contract
    </WrapperLink>
```

File: app/earn/[...vault]/header/first-line/vault-title.tsx [hunk 3]
```diff
@@ -1,30 +1,51 @@
 import type { PropsWithChildren } from 'react'
-import { ExternalAction } from '../second-line/actions'
+import { useVaultContext } from '../../hooks/use-vault-context'
```
Comment (LeoVS09) [inline app/earn/[...vault]/header/first-line/vault-title.tsx:2]: this import of context, can cause loading of unnecessary stuff

**Review by LeoVS09 — APPROVED** [review]

File: app/earn/[...vault]/header/first-line/vault-title.tsx (hunk 1 above)
Comment (solovev) [reply app/earn/[...vault]/header/first-line/vault-title.tsx:18]: it doesn't for some reason 🤔

**Pull request merged**